### PR TITLE
Fixed sourcemaps are not added by default for development builds.

### PR DESCRIPTION
**Pull request by theoshallburn**
### Ticket
[BNNCUE-4180](https://jira.stibodx.com/browse/BNNCUE-4180)

Problem
* Sourcemaps are not generated by default when compiling assets for regular development.

### Description
This change introduces a set of options similar to those used for *:production, this only affects the tasks, where the execution differs for development and production.

### CODE REVIEW

**Comment by sun** [comment]
I don't think we want to introduce a new/special command for enabling sourcemaps. They should be enabled by default for non-production builds.

It's not clear to me whether this is a general problem on all sites using gulp-task-collection currently or whether the BNN (CUE front) repo might just be lacking some required configuration.

**Comment by theoshallburn** [comment]
Regarding @sun 's last comment: 
The change does not introduce a new command, it simply expands the already existing ones by making sure running `gulp` always creates sourcemaps (styles and scripts are affected by this i believe.)

The current change adds a default configuration for the gulp task `build` similar to `build:production` , but instead of forcing `minify`, it has `sourcemaps` activated by default. Although that might not be enough if we wanna differ between dev/TEST and STAGE/PROD. If this change is wrong and the default configuration should not be changed (default as in: all options are false), it might be possible to just adapt the npm tasks on our end, so that deployment to STAGING and PROD use a different task than that to TEST or the local dev system. That would leave the code on this end unchanged, but from what @sun told me, the default should be sourcemaps on. I could not find anything where something was changed here, apparently a missing setting is automatically evaluated as falsy and thus not activated. Maybe something in the way, the options are parsed, changed, for example that non-existing values are interpreted differently, though that seems a weird change as it would definitely break backwards-compatibility. (Just guessing though, did not look deeper into this.)

I personally took this route because it explicitely states the non-prod-default options allowing for easy understanding and reading the code.

**Comment by colourgarden** [comment]
This change makes sense to me. We already have `productionDefaults` so why not `developmentDefaults`.

My only suggestion would be to make sure that `developmentDefaults` is the first block to maintain the correct order (the dev task is listed first in the block below).

As an additional note, I'd recommend to always use `npm` scripts where `gulp-task-collection` is included rather than running the available commands here directly.

File: lib/registerTaskWithProductionMode.js
Comment (colourgarden) [inline lib/registerTaskWithProductionMode.js:7]: Please move this block above the production one above.

**Comment by wrux** [comment]
I think that the changes from #33 cover this

**Comment by theoshallburn** [comment]
> I think that the changes from #33 cover this

Are you sure? I looked at that PR and this explicitely states, that `sourcemaps: false` is the default.
https://github.com/netzstrategen/gulp-task-collection/pull/33#discussion_r460199594

File: lib/registerTaskWithProductionMode.js
Comment (theoshallburn) [reply lib/registerTaskWithProductionMode.js:7]: I will, good idea!

**Comment by sun** [comment]
@theoshallburn I think we can resolve and merge this PR very quickly if you update the code as requested; but just so you're aware: you can also set custom options for the styles task here: https://jira.stibodx.com/stash/projects/BNNCUE/repos/bnncue-front-publication/browse/templates/package.json#87-89 - no need to do that now, we can update gulp-task-collection after merging this.

Let's ensure to bump the patch version as part of this PR, so we can create a new release.

File: README.md
Comment (sun) [inline README.md:41]:
```suggestion
`--sourcemaps` | Add source mappings to compiled files. Defaults to `true` in non-production builds.
```

**Review by sun — APPROVED** [review]

File: README.md
Comment (sun) [inline README.md:60]:
```suggestion
- the option `--sourceMaps` is enabled by default for development / non-production builds.

```

File: README.md
Comment (sun) [inline README.md:60]:
```suggestion
- the option `--sourcemaps` is enabled by default for development / non-production builds.
```

**Review by sun — APPROVED** [review]

**Pull request merged**
